src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1992,28 +1990,74 @@ static bool updateContextAttributeItem
   // given that the logic in NGSIv2 for specifying location attributes is gogint to change
   // (the best moment to address this FIXME is probably once NGSIv1 has been deprecated and
   // removed from code)
-  if (targetAttr->getLocation().length() > 0 && targetAttr->name != locAttr)
-  {
-    cerP->statusCode.fill(SccInvalidParameter,
-                          std::string("action: UPDATE") +
-                          " - entity: [" + eP->toString() + "]" +
-                          " - offending attribute: " + targetAttr->toString() +
-                          " - location nature of an attribute has to be defined at creation time, with APPEND");
-
-    alarmMgr.badInput(clientIp, "location nature of an attribute has to be defined at creation time, with APPEND");
+  std::string locationString = targetAttr->getLocation(apiVersion);
+
+  /* Check that location (if any) is using the correct coordinates string (it only
+   * makes sense for NGSIv1, this is legacy code that will be eventually removed) */
+  if ((locationString.length() > 0) && (locationString != LOCATION_WGS84) && (locationString != LOCATION_WGS84_LEGACY))
+  {
+    cerP->statusCode.fill(
+          SccInvalidParameter,
+          std::string("action: UPDATE") +
+          " - entity: [" + eP->toString() + "]" +
+          " - offending attribute: " + targetAttr->toString() +
+          " - only WGS84 is supported for location, found: [" + targetAttr->getLocation() + "]");
+    alarmMgr.badInput(clientIp, "only WGS84 is supported for location");
     return false;
   }
 
-  if (locAttr == targetAttr->name)
+  /* Case 1: update attribute from no-location -> location. There are 2 sub-cases */
+  if (locationString.length() > 0)
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-orion/commit/a8d0650bc335b0e735f4c57a87096bbbede7507b